artifacts/scripts/util.sh

```diff
@@ -921,7 +921,7 @@ checkout-deps-to-kube-commit() {
             git checkout -q "${dep_commit}"
 
             echo "Downloading go mod dependencies..."
-            GO111MODULE=on GOPROXY=https://proxy.golang.org go mod download
+            GO111MODULE=on GOPRIVATE="${base_package}/*" GOPROXY=https://proxy.golang.org go mod download
```

Comment:
will this break dependencies on things like k8s.io/utils?

Reply:
I think it won't break, but it'll skip validating the checksums for it.

I think this approach would be better - we have all dependencies (repo+branch name) in `deps`. We can then use `base_package` + repo names to define GOPRIVATE as an explicit list of all staging repo dependencies.

/hold
I'll address the reviews and verify that this works as expected locally

Reply:
> I think this approach would be better - we have all dependencies (repo+branch name) in `deps`. We can then use `base_package` + repo names to define GOPRIVATE as an explicit list of all staging repo dependencies.

That does sound better. Targeted and handled internally